src/Ocelot/Logging/OcelotDiagnosticListener.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DiagnosticAdapter;
+
+namespace Ocelot.Logging
+{
+    public class OcelotDiagnosticListener
+    {
+        [DiagnosticName("Microsoft.AspNetCore.MiddlewareAnalysis.MiddlewareStarting")]
+        public virtual void OnMiddlewareStarting(HttpContext httpContext, string name)
+        {
+            Console.WriteLine($"MiddlewareStarting: {name}; {httpContext.Request.Path}");

```

Comment:
Gah! I knew I forgot to do something (only the most important thing). I shall make changes and push them up later on.